tests/e2e/bento_server_http/tests/test_meta.py

```diff
@@ -25,15 +27,26 @@ async def test_api_server_meta(host: str) -> None:
     assert b'{"Hello":"World"}' == body
     status, _, _ = await async_request("GET", f"http://{host}/docs.json")
     assert status == 200
-    status, _, _ = await async_request("GET", f"http://{host}/readyz")
-    assert status == 200
     status, _, body = await async_request("GET", f"http://{host}/metrics")
     assert status == 200
     assert body
     status, _, body = await async_request("POST", f"http://{host}//api/v1/with_prefix")
     assert status == 404
 
 
+@pytest.mark.asyncio
+async def test_runner_readiness(host: str) -> None:
+    timeout = 300
+    start_time = time.time()
+    status = ""
+    while (time.time() - start_time) < timeout:
+        status, _, _ = await async_request("GET", f"http://{host}/readyz")
+        await asyncio.sleep(20)
+        if status == 200:
+            break
+    assert status == 200
```

Comment:
I think you can also test the exception by just sending a request right after, and see whether it returns 500

I believe you don't have to do a while loop, shouldn't asyncio.sleep be sufficient enough?

Reply:
@aarnphm Right, so this is the part that's puzzling me. Seoyeon and I discussed adding a while loop because with asyncio.sleep alone, the request still returns 500. Even after adding the while loop, it still returns 500 meaning the timeout of 300 was reached. I wonder if there's something preventing all the runners from becoming ready.